### PR TITLE
Shared Fedimint-UI workspace : Part 1 / 3

### PR DESCRIPTION
**Pull request by okjodom**
Introduces a shared workspace for building Fedimint UI projects.

**In this PR**

- [x] - Create a shared workspace that uses Turbo for workspace management with Yarn as the package manager. Workspace is named `fedimint-ui`
- [x] - Create shared `eslint-config-custom` and `tsconfig` libs in `fedimint-ui/packages/`
- [x] - Migrate `guardian-ui` to `fedimint-ui/apps/guardian-ui`

**Next**

- [ ] - Migrate gateway UI from `gateway/ui` to `fedimint-ui/apps/gateway-ui`
- [ ] - Create a shared `ui` library in `fedimint-ui/packages/`

See #2429

### CODE REVIEW

File: fedimint-ui/packages/tsconfig/react-app.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "jsx": "react-jsx",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "module": "esnext",
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitAny": true,
+    "noEmit": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "resolveJsonModule": true,
+    "strict": false,
```
Comment (wbobeirne) [inline fedimint-ui/packages/tsconfig/react-app.json:20]: Any reason to turn strict off for React projects?

Comment (okjodom) [reply fedimint-ui/packages/tsconfig/react-app.json:20]: none. I've now dropped the rule

File: fedimint-ui/apps/guardian-ui/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "guardian-ui",
```
Comment (wbobeirne) [inline fedimint-ui/apps/guardian-ui/package.json:2]: Should we namespace all this repo's packages with `@fedimint/*`? Maybe we should reserve that npm organization now too?

Comment (wbobeirne) [reply fedimint-ui/apps/guardian-ui/package.json:2]: This is especially valuable for package name collisions, e.g. `tsconfig` is a match for https://www.npmjs.com/package/tsconfig which could be confusing.

**Review by wbobeirne — APPROVED** [review]
Really nicely done & documented, and I like splitting this out into multiple PRs. Just had the one thought about npm package names.

File: fedimint-ui/apps/guardian-ui/package.json (hunk 2 above)
Comment (okjodom) [reply fedimint-ui/apps/guardian-ui/package.json:2]: ah great idea. agreed on namespacing

Comment (okjodom) [reply fedimint-ui/apps/guardian-ui/package.json:2]: reserved https://www.npmjs.com/org/fedimint

Comment (okjodom) [reply fedimint-ui/apps/guardian-ui/package.json:2]: I'll make a follow up PR that nameplates these packages

**Review by justinmoon — APPROVED** [review]
Looks good. Are we allowed to get rid of the `fedimint-ui/apps` intermediate directory? Just having `fedimint-ui/guardian-ui` and `fedimint-ui/gateway-ui` would make for easier `cd`'ing.

**Comment by okjodom** [comment]
> Looks good. Are we allowed to get rid of the `fedimint-ui/apps` intermediate directory? Just having `fedimint-ui/guardian-ui` and `fedimint-ui/gateway-ui` would make for easier `cd`'ing.

we need `apps/*` so we can declare that as a workspace. See this declaration in `fedimint-ui/package.json`

```json
  "packageManager": "yarn@1.22.19",
  "workspaces": [
    "apps/*",
    "packages/*"
  ]
  ```

**Comment by justinmoon** [comment]
> However, what we have now is the recommended Turborepo workspace setup

Cool. Let's just go with their recommendation for now.!

**Pull request merged**